src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -91,7 +97,14 @@ public FileSystemResourceStore(File resourceDirectory) {
     
     public synchronized void addListener(File file, String path, ResourceListener listener) {
         if( watcher == null ){
+            // Would be better if we could make FileSystemResource a DisposableBean instead
+            Disposer disposer = GeoServerExtensions.bean(Disposer.class);
```

Comment:
I thought about it but I try to resist adding static methods that return complex objects due to the problems they make for unit testing.  You are right this probably justifies itself though.